pkg/controller/clone-controller.go

```diff
@@ -158,7 +167,6 @@ func (cc *CloneController) processPvcItem(pvc *v1.PersistentVolumeClaim) error {
 	//add the following annotation only if the pod pahse is succeeded, meaning job is completed
 	if phase == string(v1.PodSucceeded) {
 		anno[AnnCloneOf] = "true"
-		defer cc.deleteClonePods(sourcePod.Namespace, sourcePod.Name, targetPod.Name)
```

Comment:
I don't see how this change ensures that the "cloneof" annotation gets applied.  It just changes the order of operations from delete pods/update pvc to update pvc/delete pods.  The controller still needs to see the pod enter "succeeded" state for the annotation to be applied.

In a separate but possibly related issue, I think it is a mistake for to always initialize `anno` to an empty map here: https://github.com/kubevirt/containerized-data-importer/blob/64cc6e7068dd29dd45db8c3ad3966b1beb885c73/pkg/controller/clone-controller.go#L92

I wonder if it is possible that the controller is actually setting the annotation then removing it when the pod enters "terminating" state.   That is just speculation.  But a definite issue is that we may be precluding users from adding their own annotations to the PVC.

By suggestion is to simply change line 92 to `anno := pvc.Annotations` and see if that fixes the issue.  If not, something else is at play.

Reply:
@mhenriks I had the same challenge, but it sounded like the if condition was sometimes causing this to be missed?  By skipping that if clauses an deferring until exit of the function it seems to eliminate the race and deletion of the pod "before" the annotation is updated.

So to make sure my interpretation matches:
1. Pod goes to completed status
2. Pod gets deleted
3. Controller looks for pod and it's status to make annotation on PVC
4. Pod no longer exists so we never update the PVC

Although... what I'm not sure is how the deferred delete call always being called, vs in the if block (which is also deferred) changes this.

There is likely a better fix.

Reply:
@j-griffith the controller deletes the pods and updates the pvc in the scope of the same function.  I don't see where your step 3 happens.